doc/move_group_interface/src/move_group_interface_tutorial.cpp

```diff
@@ -1,394 +1,648 @@
+    visual_tools.deleteAllMarkers();
+    visual_tools.publishText(text_pose, "MoveGroupInterface Demo", rvt::WHITE, rvt::XLARGE);
+    // Batch publishing is used to reduce the number of messages being sent to RViz for large visualizations
+    visual_tools.trigger();
+    // END_SUB_TUTORIAL
+  }
+
+  void print_basic_info()
+  {
+    // BEGIN_SUB_TUTORIAL basic_info
+    // Getting Basic Information
+    // ^^^^^^^^^^^^^^^^^^^^^^^^^
+    //
+    // Move_group holds a lot of information about the robot. Accessing it is easy.
+    // For example we can print the name of the reference frame for this robot.
+    ROS_INFO_NAMED("tutorial", "Planning frame: %s", move_group.getPlanningFrame().c_str());
+
+    // We can also print the name of the end-effector link for this group.
+    ROS_INFO_NAMED("tutorial", "End effector link: %s", move_group.getEndEffectorLink().c_str());
+
+    // We even can get a list of all the groups in the robot:
+    ROS_INFO_NAMED("tutorial", "Available Planning Groups:");
+    std::copy(move_group.getJointModelGroupNames().begin(), move_group.getJointModelGroupNames().end(),
+              std::ostream_iterator<std::string>(std::cout, ", "));
+
+    // Start the Demo
+    // ^^^^^^^^^^^^^^
+    visual_tools.prompt("Press 'next' in the RvizVisualToolsGui window to start the demo");
+    // END_SUB_TUTORIAL
+  }
+
+  void plan_joint_space_goal()
+  {
+    // Update the text in RViz
+    visual_tools.deleteAllMarkers();
+    visual_tools.publishText(text_pose, "Joint Space Goal", rvt::WHITE, rvt::XLARGE);
+    visual_tools.trigger();
+
+    // BEGIN_SUB_TUTORIAL joint_space
+    // Planning to a joint-space goal
+    // ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    //
+    // One of the easiest ways to move the robot is by adjusting the joint angles. To start, we'll
+    // create an pointer that references the current robot's state. RobotState is the object that
+    // contains all the current position/velocity/acceleration data.
+    moveit::core::RobotStatePtr current_state = move_group.getCurrentState();
+    //
+    // Next get the current set of joint values for the group.
+    std::vector<double> joint_group_positions;
+    current_state->copyJointGroupPositions(joint_model_group, joint_group_positions);
+
+    // Now, let's modify one of the joints, plan to the new joint space goal, and visualize the
+    // plan.
+    joint_group_positions[0] = -1.0;  // radians
+    move_group.setJointValueTarget(joint_group_positions);
+    bool success = (move_group.plan(my_plan) == moveit::planning_interface::MoveItErrorCode::SUCCESS);
+    // END_SUB_TUTORIAL
+
+    ROS_INFO_NAMED("tutorial", "Visualizing plan 1 (joint space goal) %s", success ? "" : "FAILED");
+
+    // Visualize the plan in RViz
+    visual_tools.publishTrajectoryLine(my_plan.trajectory_, joint_model_group);
+    visual_tools.trigger();
+    visual_tools.prompt("Press 'next' in the window to move to the joint space goal");
+  }
+
+  void plan_pose_goal()
```

Comment:
This should take in target_pose and return the plan as well as success/failure. Execution should be done separately. Make things modular and treat this more like a high level class for interacting with move_group.